tests/e2e/pipeline.go

```diff
@@ -441,40 +461,42 @@ func deletePipelineRollout(name string) {
 }
 
 // update PipelineRollout and verify correct process
-func updatePipelineRollout(name string, newSpec numaflowv1.PipelineSpec, expectedFinalPhase numaflowv1.PipelinePhase, dataLoss bool, overrideSourceVertexReplicas bool) {
+func UpdatePipelineRollout(name string, newSpec numaflowv1.PipelineSpec, expectedFinalPhase numaflowv1.PipelinePhase, verifySpecFunc func(numaflowv1.PipelineSpec) bool, dataLoss bool, overrideSourceVertexReplicas bool) {
 
-	document("Updating Pipeline spec in PipelineRollout")
+	Document("Updating Pipeline spec in PipelineRollout")
 	rawSpec, err := json.Marshal(newSpec)
 	Expect(err).ShouldNot(HaveOccurred())
 
 	// update the PipelineRollout
-	updatePipelineRolloutInK8S(Namespace, name, func(rollout apiv1.PipelineRollout) (apiv1.PipelineRollout, error) {
+	UpdatePipelineRolloutInK8S(Namespace, name, func(rollout apiv1.PipelineRollout) (apiv1.PipelineRollout, error) {
 		rollout.Spec.Pipeline.Spec.Raw = rawSpec
 		return rollout, nil
 	})
 
 	// no data loss OR system pause
-	if upgradeStrategy == config.PPNDStrategyID && dataLoss {
-
-		document("Verify that in-progress-strategy gets set to PPND")
-		verifyInProgressStrategy(name, apiv1.UpgradeStrategyPPND)
-
-		verifyPipelinePaused(Namespace, name)
+	if UpgradeStrategy == config.PPNDStrategyID && dataLoss {
+
+		Document("Verify that in-progress-strategy gets set to PPND")
```

Comment:
this no longer pertains to all below

Reply:
This one shouldn't be resolved. To clarify, for the "Failed" case below, we don't set the in-progress-strategy to PPND.

Reply:
Can you see if the individual functions below have `Document` statements in them? That may be sufficient then so maybe we don't need one up at the top.

Reply:
Ah I resolved since I had changed it on my working branch (using the PR as a checklist essentially) but that would probably be better to do after updating on Github. Regardless, let me check about the individual funcs and we can change that.